### ISSUE TITLE
Colonial factions spawned prematurely by Nexerelin's faction respawning system

### ISSUE THREAD

**Issue by PreConceptor** [RkShaRkz/Starsector-IntelliJ-Template#44]
### Mod Name:
Vayra Merged

### Mod Version:
3.3.10

### What did you expect to happen?
Colonial factions would start spawning via their colonization events around Cycle 212 so they can actually establish themselves.

### What actually happened?
They started spawning in late Cycle 206 around the end of September, seemingly as a result of Nexerlin thinking they are deceased and bringing them into existence, no announcement or anything. Since they had no planets before being born prematurely they pick a random planet, usually terrible, and either struggle along pathetically or just die again very quickly.

By using the console command string below that prevents the 5 colonial factions respawning, they stop appearing prematurely. That's how I'm 100% sure its a Nexerelin faction respawn issue.

runcode Global.getSector().getFaction("almighty_dollar").getMemoryWithoutUpdate().set("$nex_respawn_cooldown", true);
		Global.getSector().getFaction("ashen_keepers").getMemoryWithoutUpdate().set("$nex_respawn_cooldown", true);
		Global.getSector().getFaction("communist_clouds").getMemoryWithoutUpdate().set("$nex_respawn_cooldown", true);
		Global.getSector().getFaction("science_fuckers").getMemoryWithoutUpdate().set("$nex_respawn_cooldown", true);
		Global.getSector().getFaction("warhawk_republic").getMemoryWithoutUpdate().set("$nex_respawn_cooldown", true);. 

Also noticed that when this command string is used, Fays Co from Bird's Trinkets and Rarities (which is disabled via Lunalib) tries to do a Resurgence at exactly the same time a colonial faction would sneak in. That actually pops up a notification, so even more evidence its a wonky interaction with the Nex system.

This also happened with older versions of Vayra's Sector, even since before Dragn was maintaining it.

### Steps to Reproduce:
1. Turn on Nexerelin's faction respawning setting for all factions and wait, for me it first happened on Sept 26 c206. At least one of the colonial factions will be present in the faction list with no announcement and either have no planet or a terrible one, or maybe just vanish again after a while.

2. Using the above command string to flag the colonial factions not to respawn prevents this.

**Comment by RkShaRkz** [comment]
@PreConceptor ok, i see that the issue reporting template is missing the `"What should have happened"` question since i'm not exactly sure what... needs fixing here.

So, nexerelin is causing problems with these factions, and you ran code via console to make them show up 

So... what exactly needs fixing here? How these factions work with nexerelin? The Nexerelin integration? 
can you clarify this a bit better to me on discord please? or here?

**Comment by PreConceptor** [comment]
@RkShaRkz 'What should have happened?' and 'What did you expect to happen?' are identical in this case, and the answer is 'not spawn 6 months into the first cycle in such a way that effectively renders the factions stillborn without having to input a specific console command that 99.9% of users won't know'. 

I only know it because I noticed this problem several years ago and figured it out by asking around on discord, it was not and still isn't a commonly known solution. What needs fixing is also pretty obvious, although I don't know if its possible to fix only here. Something that prevents Nexerelin thinking the factions are already dead and need to be respawned, whether a custom thing or just flagging them on game start to not respawn so the respawning system doesn't bypass their intended spawning via Vayra's Sector-controlled expeditions. I suppose the other option is to put a clear warning somewhere to not enable Nex faction respawning with Vayra Merged if you don't want to deal with the bug.

**Comment by RkShaRkz** [comment]
Histidine chimed in

> call `SectorManager.getManager().incrementNumRespawns(factionId)` 99 times
> (at such a point I'd seriously consider using reflection to set the large value directly :death:)
> actually lemme see if there's a more dynamic way to do it, e.g. letting the factions respawn normally after they've scripted-spawned once
> oh `SectorManager.DO_NOT_RESPAWN_FACTIONS` is a static set, you can manage it yourself

**Comment by RkShaRkz** [comment]
so the thing i'm kinda-but-not-really worried about now (that i need to recheck) is - using Nexerelin's SectorManager directly will put in a hard-dependancy on nexerelin rather than it being an optional one. so I need to think a bit about that, ideally keeping it optional but also playing around with that if it's present :) 

either way, the following factions need to be added to the nexerelin's `SectorManager.DO_NOT_RESPAWN_FACTIONS`
```
communist_clouds
science_fuckers
warhawk_republic
almighty_dollar
ashen_keepers
kadur_remnant
```

**Comment by RkShaRkz** [comment]
@PreConceptor so, if i upload a build for you to test out before i merge in the "fix", would you be able to try it out and confirm whether it works or not so the proper fix can be merged in? :) i'm rksharkz on discord. otherwise doing it by email is no different from doing it by using this issue as a mediator.

**Comment by RkShaRkz** [comment]
build with potential fix for issue #44 - https://ufile.io/ie0swkbc

**Comment by RkShaRkz** [comment]
Resolved and fixed by commit 84caff3 - closing.
